pkg/nodeconfig/nodeconfig.go

```diff
@@ -244,9 +245,13 @@ func (nc *nodeConfig) Configure() error {
 	return err
 }
 
-// safeReboot safely restarts the instance associated with the given node object, first cordoning and draining the node
-func (nc *nodeConfig) SafeReboot(ctx context.Context, node *core.Node) error {
-	nc.node = node
+// safeReboot safely restarts the underlying instance, first cordoning and draining the associated node.
+// Waits for reboot to take effect before uncordoning the node.
+func (nc *nodeConfig) SafeReboot(ctx context.Context) error {
+	if nc.node == nil {
+		return fmt.Errorf("safe reboot of the instance requires an associated node")
+	}
```

Comment:
Consider adding a unit test with the new signature of the function to cover the edge case where `nc.node == nil`, and an error is thrown.

Reply:
IMO the work required to mock the clients and resources needed to unit test this func is too costly compared to the benefit of the unit test